sqlmesh/core/loader.py

```diff
@@ -384,15 +389,42 @@ def _raise_failed_to_load_model_error(self, path: Path, error: t.Union[str, Exce
 class SqlMeshLoader(Loader):
     """Loads macros and models for a context using the SQLMesh file formats"""
 
+    @property
+    def is_migrated_dbt_project(self) -> bool:
+        return self.migrated_dbt_project_name is not None
+
+    @property
+    def migrated_dbt_project_name(self) -> t.Optional[str]:
+        return self.config.variables.get(c.MIGRATED_DBT_PROJECT_NAME)
```

Comment:
Since we know from the config that this a migrated project, can we put all of the logic below into a separate loader that inherits from `SqlMeshLoader`?